### PR TITLE
Delete page entries from db for single url.

### PR DESCRIPTION
**Pull request by wimpykid26**
@Treora What I did was:
Used findpagesbyurl and normalizeresults from deduplication branch.
get the corresponding entries
called db.remove

Please review. If there is any error I'll fix them ASAP.

### CODE REVIEW

**Comment by wimpykid26** [comment]
@Treora any thoughts?

**Comment by Treora** [comment]
It looks like this PR also contains the bookmarks import code and other stuff that is not related to this issue, so the diff is too messy to be reviewed.
By the way, I merged deduplication into master, so if you rebase onto master you can also use the `findPagesByUrl`.

**Comment by wimpykid26** [comment]
@Treora Apologize for the mess. Something went wrong while creating the branch. cleared the diff. used `findPagesByUrl` to get corresponding entries of same url.

File: src/delete-pages/delete-pages.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      return updatePageSearchIndex()
+    }).then(() => {
+      console.timeEnd('rebuild search index')
+    }).catch(err => console.log(err));
+  }
+}
+
+export default function deletePage(query) {
+  console.log(query)
+  var result = findPagesByUrl({query, limit: defaultResultLimit}).then(
```
Comment (Treora) [inline src/delete-pages/delete-pages.js]: `findPagesByUrl` does not take a `query` or `limit` argument.. how has this worked for you at all?

File: src/delete-pages/delete-pages.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import db from '../pouchdb'
+import { findPagesByUrl, updatePageSearchIndex } from '../search/find-pages'
+
+const defaultResultLimit = 30
```
Comment (Treora) [inline src/delete-pages/delete-pages.js]: Why a result limit?

File: src/delete-pages/delete-pages.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+//Delete all Page entries returned by findPagesByUrl
+function deletePageEntries(pagesResult) {
+  console.log(pagesResult);
+  const pages = pagesResult.rows.map(row => ({
+    id: row.doc._id,
+    rev: row.doc_rev
+  }));
+  for (var key in pages) {
+    db.get(pages[key].id).then(doc => db.remove(doc._id, doc._rev)).then(() => {
```
Comment (Treora) [inline src/delete-pages/delete-pages.js]: Why fiddle around with `id`s and `rev`s and get each page again? Just running `pagesResult.rows.map(row => db.remove(row.doc))` should work, no?

File: src/delete-pages/delete-pages.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      console.timeEnd('Deleting pages end')
+      console.time('rebuild search index')
+      return updatePageSearchIndex()
```
Comment (Treora) [inline src/delete-pages/delete-pages.js]: You don't want to reindex after every single deletion. Perhaps use `Promise.all` on the return value of the pages.
Or, equivalently, replace the promises with an `async function` that `await`s each removal.

**Review by Treora — COMMENTED** [review]
That is a lot easier to oversee indeed. The code looks rather broken though; has it even worked for you? Please first get things to work before asking for review.

**Comment by wimpykid26** [comment]
screenshot from 2017-03-21 01-39-12
@Treora Cleaned the code as you said. Made the pagesResult return value into a promise and rebuilded search index after the promise is returned.

**Comment by Treora** [comment]
The essence is not bad, but it's still only half the solution here; closing due to inactivity.

**Pull request closed without merging**